clang/tools/clang-cache-build-session/CMakeFileAPI.cpp

```diff
@@ -0,0 +1,228 @@
+//===-- CMakeFileAPI.cpp - CMake File API ---------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "CMakeFileAPI.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/Path.h"
+
+using namespace llvm;
+
+/// \returns true if \p Path is a nested directory in \p ParentPath.
+/// \p Path should be absolute.
+static bool isPathNestedIn(StringRef Path, StringRef ParentPath) {
+  assert(sys::path::is_absolute(Path));
+  if (Path.size() <= ParentPath.size())
+    return false;
+  if (!Path.startswith(ParentPath))
+    return false;
+  return Path.drop_front(ParentPath.size())
+      .startswith(sys::path::get_separator());
```

Comment:
We should use `sys::path::is_separator` instead of `get_separator` since there are multiple valid separators on Windows.  And since you already checked ParentPath.size() > Path.size() you know there is at least one character to check.

Reply:
Good point, done 👍